File: backend/src/main/java/de/ropold/backend/model/UserModel.java
package de.ropold.backend.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "users")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class UserModel {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(columnDefinition = "UUID")
    private UUID id;

    @Column(name = "github_id", nullable = false, unique = true)
    private String githubId;

    @Column(name = "username", nullable = false, unique = true)
    private String username;

    @Column(name = "name", nullable = false)
    private String name;

    @Column(name = "avatar_url")
    private String avatarUrl;

    @Column(name = "github_url")
    private String githubUrl;

    @Column(name = "role", nullable = false)
    private String role = "USER";

    @Column(name = "preferred_language", length = 2)
    private String preferredLanguage = "de";

    @Column(name = "created_at", nullable = false)
    private LocalDateTime createdAt;

    @Column(name = "last_login_at")
    private LocalDateTime lastLoginAt;
}

File: frontend/src/components/models/UserModel.ts

export type UserModel = {
    id: string;
    githubId: string;
    username: string;
    name: string;
    avatarUrl?: string;
    githubUrl?: string;
    role: string;
    preferredLanguage: string;
    createdAt?: string;
    lastLoginAt?: string;
};

export const DefaultUser: UserModel = {
    id: "0",
    githubId: "0",
    username: "Loading...",
    name: "Loading...",
    role: "USER",
    preferredLanguage: "de",
    avatarUrl: undefined,
    githubUrl: undefined,
    lastLoginAt: undefined,
    createdAt: "Loading..."
}